rnd/autogpt_builder/Dockerfile

```diff
@@ -0,0 +1,32 @@
+# Base stage for both dev and prod
+FROM node:21-alpine AS base
+WORKDIR /app
+COPY autogpt_builder/package*.json ./
+RUN npm ci
+
+# Dev stage
+FROM base AS dev
+ENV NODE_ENV=development
+COPY autogpt_builder/ .
+EXPOSE 3000
+CMD ["npm", "run", "dev"]
+
+# Build stage for prod
+FROM base AS build
+COPY autogpt_builder/ .
+RUN npm run build
+
+# Prod stage
+FROM node:21-alpine AS prod
+ENV NODE_ENV=production
+WORKDIR /app
+
+COPY --from=build /app/package*.json ./
+RUN npm ci --only=production
```

Comment:
we do have a package-lock.json, it doesnt get checked in, there's no need for both package and package-lock

Reply:
If it doesn't get checked in, that means we don't have one. Also, the purpose of such lock files (in this case) is to make builds reproducible and prevent implicit, unintended package version changes. Furthermore, `npm ci` fails if `package-lock.json` isn't present, which is why I suggested using `yarn` because we do have a `yarn.lock` and it doesn't make sense to use two different lock files.

Reply:
I understand lock files

> npm ci
fails, this I didn't know, thanks

re yarn, when it was introduced, should have also gone ahead and replaced npm in the codebase, given it wasn't lets do the remaining in a follow up PR